jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentScanner.java

```diff
@@ -301,20 +272,56 @@ void addScannerListener(Scanner.Listener listener)
         scanner.addListener(listener);
     }
 
+    protected int getDefaultWeight(String name)
+    {
+        URL url = getClass().getResource("environment-weights.properties");
+        if (url == null)
+            throw new IllegalStateException("Unable to find environment-weights.properties in classloader");
+        try (InputStream in = url.openStream())
+        {
+            Properties props = new Properties();
+            props.load(in);
+            String value = props.getProperty(name);
+            if (value == null)
+                throw new IllegalStateException("Unknown environment [%s], unable to find default weight, use configureEnvironment(name, weight) instead."
+                    .formatted(name));
+            return Integer.parseInt(value);
+        }
+        catch (IOException e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
     /**
      * Configure the Environment specific Deploy settings.
      *
      * @param name the name of the environment.
      * @return the deployment configuration for the {@link Environment}.
      */
     public EnvironmentConfig configureEnvironment(String name)
+    {
+        return configureEnvironment(name, getDefaultWeight(name));
+    }
+
+    /**
+     * Configure the Environment specific Deploy settings.
+     *
+     * @param name the name of the environment.
+     * @param weight the weight for environment selection when calculating a default environment.
+     *    (Example usage: {@code static} is 100, {@code core} is 200, {@code ee8} is 1008, {@code ee11} is 1011)
+     * @return the deployment configuration for the {@link Environment}.
+     */
+    public EnvironmentConfig configureEnvironment(String name, int weight)
```

Comment:
Just have this method and delete `configureEnvironment(String name)`.
Rename this to `trackEnvironment()` maybe?
The XML files will pass in the weight and there will be no need for the `environment-weights.properties` file.
This method is only called by tests or XML files.